.github/workflows/deploy_docker_image.yml

```diff
@@ -0,0 +1,86 @@
+name: Deploy Harmony GDAL Adapter Docker image
+
+on:
+  push:
+    branches: [ main ]
+
+env:
+  IMAGE_NAME: {{ github.repository }}
+  REGISTRY: ghcr.io
+
+jobs:
+  run_tests:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ 3.8 ]
+
+    steps:
+      - name: Checkout harmony-gdal-adapter repository
+        uses: actions/checkout@v2
+        with:
+          lfs: true
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Build test image
+        run: ./bin/build-test
+
+      - name: Run test image
+        run: ./bin/run-test
+
+      - name: Archive test results
+        uses: actions/upload-artifact@v2
+        with:
+          name: Test results for Python ${{ matrix.python-version }}
+          path: test-reports/
+
+      - name: Archive coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: Coverage report for Python ${{ matrix.python-version }}
+          path: coverage/*
+
+  build_and_deploy_image:
+    needs: run_tests
```

Comment:
The `needs` parts of the GitHub jobs syntax is neat. Using it here means that the `build_and_deploy_image` job can't run unless the `run_tests` job has already run (and completed successfully).